src/mofaflex/_core/mofaflex.py

```diff
@@ -550,13 +567,46 @@ def _setup_gp(self, covariates=None, full_setup=True):
             self._gp_group_names = None
         return gp_warp_groups_order
 
-    def _setup_svi(self, prior_scales, init_tensor, covariates, feature_means, sample_means):
+    def _setup_guiding_vars(self):
+        self._guiding_vars_names = (
+            list(self._data_opts.guiding_vars_obs_keys.keys()) if self._data_opts.guiding_vars_obs_keys else []
+        )
+        self._n_guiding_vars = len(self._guiding_vars_names)
+        self._guiding_vars_factors = {}
+        self._guiding_vars_n_categories = {}
```

Comment:
I still don't like the idea of storing things in the MOFAFLEX object that are only used during training and never again, this sort of violates the idea of the class being a results storage class. Is there no way to pass this information by returns and function arguments, like with other things (e.g. `init_tensor` or `prior_scales`)?